Doctrine/Phpcr/SimpleUrlInformationGuesser.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2015 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\SeoBundle\Doctrine\Phpcr;
+
+use Symfony\Cmf\Bundle\SeoBundle\AlternateLocaleProviderInterface;
+use Symfony\Cmf\Bundle\SeoBundle\Model\UrlInformation;
+use Symfony\Cmf\Bundle\SeoBundle\SeoPresentation;
+use Symfony\Cmf\Bundle\SeoBundle\Sitemap\UrlInformationGuesserInterface;
+use Symfony\Component\Routing\RouterInterface;
+
+/**
+ * A base guesser to extract url information from a document persisted with the phpcr-odm.
+ *
+ * @author Maximilian Berghoff <Maximilian.Berghoff@gmx.de>
+ */
+class SimpleUrlInformationGuesser implements UrlInformationGuesserInterface
+{
+    /**
+     * @var AlternateLocaleProviderInterface
+     */
+    protected $alternateLocaleProvider;
+
+    /**
+     * @var RouterInterface
+     */
+    private $router;
+
+    /**
+     * @var string
+     */
+    private $defaultChangeFrequency;
+
+    /**
+     * @var SeoPresentation
+     */
+    private $seoPresentation;
+
+    /**
+     * @param RouterInterface $router
+     * @param SeoPresentation $seoPresentation
+     * @param $defaultChangeFrequency
+     */
+    public function __construct(
+        RouterInterface $router,
+        SeoPresentation $seoPresentation,
+        $defaultChangeFrequency
+    ) {
+        $this->router = $router;
+        $this->seoPresentation = $seoPresentation;
+        $this->defaultChangeFrequency = $defaultChangeFrequency;
+    }
+
+    /**
+     * @param AlternateLocaleProviderInterface $alternateLocaleProvider
+     */
+    public function setAlternateLocaleProvider(AlternateLocaleProviderInterface $alternateLocaleProvider)
+    {
+        $this->alternateLocaleProvider = $alternateLocaleProvider;
+    }
+
+    /**
+     * {@inheritDocs}
+     */
+    public function guessValues(UrlInformation $urlInformation, $object, $sitemap = 'default')
+    {
+        $urlInformation->setLocation($this->router->generate($object, array(), true));
```

Comment:
But @dbu would have a dependecy confusion again:

```
        $urlInformation->setLocation($this->router->generate($object, array(), true));
        $urlInformation->setChangeFrequency($this->defaultChangeFrequency);

        if ($this->alternateLocaleProvider) {
            $collection = $this->alternateLocaleProvider->createForContent($object);
            $urlInformation->setAlternateLocales($collection->toArray());
        }

        $seoMetadata = $this->seoPresentation->getSeoMetadata($object);
        if (null !== $seoMetadata->getTitle()) {
            $urlInformation->setLabel($seoMetadata->getTitle());
            return $urlInformation;
        }
``` 

would be the current code for a simple global guesser for the url information. But there are at least 2 properties, which are created by a service which depends on `PHPCR`: location (router won't work) and alternate locale. Not to set the alternate locale for a non existing service would be ok, but the location? 
At the end the complete guesser depends on `PHCR`, but we won't be able to skip the guesser itself inside of the sitemap controller, as there would be no content at all. So the controller itself depends on `PHPCR`.